src/event-manager.js

```diff
@@ -67,49 +67,53 @@ class CapturedHandlerEntry {
   }
 }
 
-function handleDelegatedEvent(event) {
-  event.propagationStopped = false;
-  let target = findOriginalEventTarget(event);
+class DelegateHandlerEntry {
+  constructor(eventName, eventManager) {
+    this.eventName = eventName;
+    this.count = 0;
+    this.eventManager = eventManager;
+  }
 
-  while (target && !event.propagationStopped) {
-    if (target.delegatedCallbacks) {
-      let callback = target.delegatedCallbacks[event.type];
-      if (callback) {
-        if (event.stopPropagation !== stopPropagation) {
-          event.standardStopPropagation = event.stopPropagation;
-          event.stopPropagation = stopPropagation;
-        }
-        if ('handleEvent' in callback) {
-          callback.handleEvent(event);
-        } else {
-          callback(event);
+  handleDelegatedEvent = event => {
```

Comment:
This means every entry, we will create a new fn for handling the event. I don't think this is the direction we would want to go.
So it looks like we need a way to associate the handler with an Entry instance, `handleEvent` seems like a perfect interface for this.

So we can have this:
```diff
class DelegateHandlerEntry {

+  handleEvent(event) {
+    handleDelegatedEvent(event, this.escapeShadowDom); // or we can just inline the existing loose function handler here
+  }
-  handleDelegatedEvent = event => {
-    ...  
}
```
And please don't forget capture too 😄

Reply:
@bigopon Changed the arrow func into a class method, as a result the handleDelegatedEvent becomes part of the prototype chain. This means only the params in the this scope will be stored in the new object, so overhead on memory is minimized now.